### PR TITLE
Clean up code that depends on unused older versions of websocket

### PR DESCRIPTION
**Pull request by amizurov**
Motivation:

We are not currently using older versions of websocket. We can remove unused code

Modification:

Remove code that depends on unused older versions of websocket.

Result:

Removed unused code.

### CODE REVIEW

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
hmm the leak looks unrelated... @chrisvest WDYT ?

```
2022-03-24T10:23:53.4340526Z 10:23:53.432 [main] INFO  i.n.t.t.socket.SocketHalfClosedTest - Running: testAllDataReadAfterHalfClosureByteBuf 1 of 1 (ServerBootstrap(ServerBootstrapConfig(group: MultithreadEventLoopGroup, channelFactory: ReflectiveServerChannelFactory(NioServerSocketChannel.class), localAddress: /127.0.0.1:0, options: {ALLOCATOR=UnpooledByteBufAllocator(directByDefault: true), BUFFER_ALLOCATOR=io.netty5.buffer.api.ManagedBufferAllocator@79ab3a71}, childGroup: MultithreadEventLoopGroup, childOptions: {ALLOCATOR=UnpooledByteBufAllocator(directByDefault: true), BUFFER_ALLOCATOR=io.netty5.buffer.api.ManagedBufferAllocator@79ab3a71})) + Bootstrap(BootstrapConfig(group: MultithreadEventLoopGroup, channelFactory: ReflectiveChannelFactory(NioSocketChannel.class), options: {ALLOCATOR=UnpooledByteBufAllocator(directByDefault: true), BUFFER_ALLOCATOR=io.netty5.buffer.api.ManagedBufferAllocator@79ab3a71}, resolver: io.netty5.resolver.DefaultAddressResolverGroup@517bd097))) with UnpooledByteBufAllocator
2022-03-24T10:23:53.4402453Z 10:23:53.436 [testsuite-nio-worker-3-2] ERROR io.netty.util.ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
2022-03-24T10:23:53.4403344Z Recent access records: 
2022-03-24T10:23:53.4403793Z Created at:
2022-03-24T10:23:53.4404415Z 	io.netty.buffer.PooledByteBufAllocator.newDirectBuffer(PooledByteBufAllocator.java:401)
2022-03-24T10:23:53.4405236Z 	io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:188)
2022-03-24T10:23:53.4406055Z 	io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:174)
2022-03-24T10:23:53.4407121Z 	io.netty.buffer.AbstractByteBufAllocator.buffer(AbstractByteBufAllocator.java:108)
2022-03-24T10:23:53.4407983Z 	io.netty5.testsuite.transport.socket.SocketCancelWriteTest$TestHandler.messageReceived(SocketCancelWriteTest.java:191)
2022-03-24T10:23:53.4408878Z 	io.netty5.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
2022-03-24T10:23:53.4409757Z 	io.netty5.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:364)
2022-03-24T10:23:53.4410695Z 	io.netty5.channel.DefaultChannelHandlerContext.findAndInvokeChannelRead(DefaultChannelHandlerContext.java:358)
2022-03-24T10:23:53.4411623Z 	io.netty5.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:339)
2022-03-24T10:23:53.4412401Z 	io.netty5.channel.ChannelHandler.channelRead(ChannelHandler.java:237)
2022-03-24T10:23:53.4413192Z 	io.netty5.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:364)
2022-03-24T10:23:53.4414138Z 	io.netty5.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:824)
2022-03-24T10:23:53.4414935Z 	io.netty5.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:187)
2022-03-24T10:23:53.4415666Z 	io.netty5.channel.nio.NioHandler.processSelectedKey(NioHandler.java:584)
2022-03-24T10:23:53.4416374Z 	io.netty5.channel.nio.NioHandler.processSelectedKey(NioHandler.java:544)
2022-03-24T10:23:53.4417121Z 	io.netty5.channel.nio.NioHandler.processSelectedKeysOptimized(NioHandler.java:525)
2022-03-24T10:23:53.4417868Z 	io.netty5.channel.nio.NioHandler.processSelectedKeys(NioHandler.java:457)
2022-03-24T10:23:53.4418520Z 	io.netty5.channel.nio.NioHandler.run(NioHandler.java:434)
2022-03-24T10:23:53.4419186Z 	io.netty5.channel.SingleThreadEventLoop.runIo(SingleThreadEventLoop.java:207)
2022-03-24T10:23:53.4419904Z 	io.netty5.channel.SingleThreadEventLoop.run(SingleThreadEventLoop.java:191)
2022-03-24T10:23:53.4420720Z 	io.netty5.util.concurrent.SingleThreadEventExecutor.lambda$doStartThread$4(SingleThreadEventExecutor.java:768)
2022-03-24T10:23:53.4421548Z 	io.netty5.util.internal.ThreadExecutorMap.lambda$apply$1(ThreadExecutorMap.java:68)
2022-03-24T10:23:53.4432680Z 	io.netty5.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2022-03-24T10:23:53.4438547Z 	java.base/java.lang.Thread.run(Thread.java:829)
2022-03-24T10:23:53.5917002Z 10:23:53.590 [main] INFO  i.n.t.t.socket.SocketHalfClosedTest - Running: testAllDataReadAfterHalfClosureByteBuf 1 of 1 (ServerBootstrap(ServerBootstrapConfig(group: MultithreadEventLoopGroup, channelFactory: ReflectiveServerChannelFactory(NioServerSocketChannel.class), localAddress: /127.0.0.1:0, options: {ALLOCATOR=PooledByteBufAllocator(directByDefault: true), BUFFER_ALLOCATOR=io.netty5.buffer.api.pool.PooledBufferAllocator@6e5bfdfc}, childGroup: MultithreadEventLoopGroup, childOptions: {ALLOCATOR=PooledByteBufAllocator(directByDefault: true), BUFFER_ALLOCATOR=io.netty5.buffer.api.pool.PooledBufferAllocator@6e5bfdfc})) + Bootstrap(BootstrapConfig(group: MultithreadEventLoopGroup, channelFactory: ReflectiveChannelFactory(NioSocketChannel.class), options: {ALLOCATOR=PooledByteBufAllocator(directByDefault: true), BUFFER_ALLOCATOR=io.netty5.buffer.api.pool.PooledBufferAllocator@6e5bfdfc}, resolver: io.netty5.resolver.DefaultAddressResolverGroup@517bd097))) with PooledByteBufAllocator
```

**Review by Kvicii — APPROVED** [review]

**Comment by amizurov** [comment]
@normanmaurer, @chrisvest Could you please check the fix of memory leak

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java [hunk 1]
```diff
@@ -509,55 +509,58 @@ private static void testResponseWithContentLengthFragmented(byte[] header, int f
...
-            assertThat(content.payload().readableBytes(), is(16));
+            assertEquals(new EmptyLastHttpContent(ch.bufferAllocator()), content);
         }
 
-        assertThat(ch.finish(), is(true));
+        assertTrue(ch.finish());
 
-        try (Buffer expected = copiedBuffer(ch.bufferAllocator(), otherData);
-             Buffer buffer = ch.readInbound()) {
+        try (Buffer expected = copiedBuffer(ch.bufferAllocator(), otherData); Buffer buffer = ch.readInbound()) {
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:563]: Line break removed by mistake?

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java [hunk 2]
```diff
@@ -509,55 +509,58 @@ private static void testResponseWithContentLengthFragmented(byte[] header, int f
...
+        assertTrue(ch.writeInbound(ch.bufferAllocator().copyOf(otherData)));
 
         HttpResponse res = ch.readInbound();
         assertThat(res.protocolVersion(), sameInstance(HttpVersion.HTTP_1_1));
-        assertThat(res.status(), is(HttpResponseStatus.SWITCHING_PROTOCOLS));
+        assertThat(res.status(), sameInstance(HttpResponseStatus.SWITCHING_PROTOCOLS));
+
         try (HttpContent<?> content = ch.readInbound()) {
-            assertThat(content.payload().readableBytes(), is(16));
+            assertEquals(new EmptyLastHttpContent(ch.bufferAllocator()), content);
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:558]: The `EmptyLastHttpContent` you build for the expected case should also be closed.

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpClientCodecTest.java [hunk 3]
```diff
@@ -322,27 +321,24 @@ public void testDecodesFinalResponseAfterSwitchingProtocols() {
...
         assertThat(res.protocolVersion(), sameInstance(HttpVersion.HTTP_1_1));
-        assertThat(res.status(), is(HttpResponseStatus.SWITCHING_PROTOCOLS));
-        HttpContent<?> content = ch.readInbound();
-        assertThat(content.payload().readableBytes(), is(16));
-        content.close();
+        assertThat(res.status(), sameInstance(HttpResponseStatus.SWITCHING_PROTOCOLS));
 
-        assertThat(ch.finish(), is(false));
+        LastHttpContent<?> lastHttpContent = ch.readInbound();
+        assertEquals(new EmptyLastHttpContent(ch.bufferAllocator()), lastHttpContent);
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty5/handler/codec/http/HttpClientCodecTest.java:339]: I think the `lastHttpContent` and `new EmptyLastHttpContent()` should be closed as well.

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpClientCodecTest.java [hunk 4]
```diff
@@ -322,27 +321,24 @@ public void testDecodesFinalResponseAfterSwitchingProtocols() {
...
+        byte[] data = ("HTTP/1.1 101 Switching Protocols\r\n" +
+                       "Upgrade: websocket\r\n" +
+                       "Connection: Upgrade\r\n" +
+                       "Sec-WebSocket-Accept: s3pPLMBiTxaQ9kYGzzhZRbK+xOo= \r\n" +
+                       "Sec-WebSocket-Protocol: chat\r\n\r\n")
+                .getBytes(CharsetUtil.US_ASCII);
         EmbeddedChannel ch = new EmbeddedChannel(new HttpClientCodec());
         assertTrue(ch.writeInbound(ch.bufferAllocator().copyOf(data)));
 
         HttpResponse res = ch.readInbound();
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty5/handler/codec/http/HttpClientCodecTest.java:334]: `HttpResponse` does not itself have a `close()` method, but pretty much everything that implements it does.
I think it'd be prudent to pass it to `Resource.dispose()` when we're done with it.

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java [hunk 5]
```diff
@@ -509,55 +509,58 @@ private static void testResponseWithContentLengthFragmented(byte[] header, int f
...
+                       "Connection: Upgrade\r\n" +
+                       "Sec-WebSocket-Accept: s3pPLMBiTxaQ9kYGzzhZRbK+xOo= \r\n" +
+                       "Sec-WebSocket-Protocol: chat\r\n\r\n")
+                .getBytes(CharsetUtil.US_ASCII);
+
         EmbeddedChannel ch = new EmbeddedChannel(new HttpResponseDecoder());
-        ch.writeInbound(ch.bufferAllocator().allocate(data.length).writeBytes(data));
+        assertTrue(ch.writeInbound(ch.bufferAllocator().copyOf(data)));
 
         HttpResponse res = ch.readInbound();
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:522]: Another `HttpResponse` that could be containing a buffer.

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java [hunk 6]
```diff
@@ -509,55 +509,58 @@ private static void testResponseWithContentLengthFragmented(byte[] header, int f
...
+        assertEquals(4, res.headers().size());
+        assertTrue(res.headers().contains(HttpHeaderNames.UPGRADE, HttpHeaderValues.WEBSOCKET, true));
+        assertTrue(res.headers().contains(HttpHeaderNames.CONNECTION, HttpHeaderValues.UPGRADE, true));
+        assertTrue(res.headers().contains(HttpHeaderNames.SEC_WEBSOCKET_ACCEPT, "s3pPLMBiTxaQ9kYGzzhZRbK+xOo=",
+                                          false));
+        assertTrue(res.headers().contains(HttpHeaderNames.SEC_WEBSOCKET_PROTOCOL, "chat", false));
 
-        assertThat(ch.finish(), is(false));
+        LastHttpContent<?> lastHttpContent = ch.readInbound();
+        assertEquals(new EmptyLastHttpContent(ch.bufferAllocator()), lastHttpContent);
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:533]: I think the `lastHttpContent` and `new EmptyLastHttpContent()` should be closed as well.

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java [hunk 7]
```diff
@@ -509,55 +509,58 @@ private static void testResponseWithContentLengthFragmented(byte[] header, int f
...
+                .getBytes(CharsetUtil.US_ASCII);
+        byte[] otherData = { 1, 2, 3, 4 };
 
         EmbeddedChannel ch = new EmbeddedChannel(new HttpResponseDecoder());
-        ch.writeInbound(copiedBuffer(ch.bufferAllocator(), data));
-        ch.writeInbound(copiedBuffer(ch.bufferAllocator(), otherData));
+        assertTrue(ch.writeInbound(ch.bufferAllocator().copyOf(data)));
+        assertTrue(ch.writeInbound(ch.bufferAllocator().copyOf(otherData)));
 
         HttpResponse res = ch.readInbound();
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:553]: Another `HttpResponse` that we could dispose when we're done with it.

File: testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketCancelWriteTest.java [hunk 8]
```diff
@@ -161,7 +165,17 @@ public void testCancelWrite(ServerBootstrap sb, Bootstrap cb) throws Throwable {
             throw ch.exception.get();
         }
         assertEquals(0, ch.counter.get());
+        assertNull(ch.received);
         assertEquals(preferredAllocator().copyOf(new byte[] { 'b', 'c', 'e' }), sh.received);
+        releaseReceivedBuffer(sh.received);
+    }
+
+    private void releaseReceivedBuffer(Object buffer) {
```
Comment (chrisvest) [inline testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketCancelWriteTest.java:173]: You can use `Resource.dispose()` which does the same thing.

**Review by chrisvest — COMMENTED** [review]
@amizurov The specific leak looks like it was plugged, but I spotted other things. Please take a look.

File: testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketCancelWriteTest.java (hunk 8 above)
Comment (amizurov) [reply testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketCancelWriteTest.java:173]: Thanks, fixed.

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java (hunk 7 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:553]: Done

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java (hunk 6 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:533]: Done

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java (hunk 5 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java:522]: Done

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpClientCodecTest.java (hunk 4 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty5/handler/codec/http/HttpClientCodecTest.java:335]: Let's, add

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java (hunk 2 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java]: Done

File: codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java (hunk 1 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java]: Fixed

**Review by normanmaurer — APPROVED** [review]

**Review by chrisvest — APPROVED** [review]

**Comment by chrisvest** [comment]
Thanks!

**Pull request merged**